vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -3100,18 +3102,26 @@ private void appendChildColumns(List<Column<T>> list,
 
     @ClientCallable
     private void select(String key) {
-        getSelectionModel().selectFromClient(findByKey(key));
+        T item = findByKey(key);
+        if (item != null) {
+            getSelectionModel().selectFromClient(item);
+        }
     }
 
     @ClientCallable
     private void deselect(String key) {
-        getSelectionModel().deselectFromClient(findByKey(key));
+        T item = findByKey(key);
+        if (item != null) {
+            getSelectionModel().deselectFromClient(item);
+        }
     }
 
     private T findByKey(String key) {
         T item = getDataCommunicator().getKeyMapper().get(String.valueOf(key));
         if (item == null) {
-            throw new IllegalStateException("Unknown key: " + key);
+            LoggerFactory.getLogger(Grid.class)
+                    .warn("Cannot select or deselect with key: " + key
```

Comment:
Should this be logged as INFO instead of WARN if you already know it's possible that this happens with frequent use? A lot of environment automatically send WARN to a central log server or create some other type of alerting

Reply:
Maybe, but I think it is not very frequent, as it took quite some time from first reporting of this issue to get it prioritized first time. But most likely there will be some internal discussion, what is the most appropriate log level.